mappings/net/minecraft/data/client/model/BlockStateModelGenerator.mapping

```diff
@@ -1,12 +1,422 @@
 	METHOD method_25681 registerSimpleState (Lnet/minecraft/class_2248;)V
 		ARG 1 block
+	METHOD method_25682 registerBed (Lnet/minecraft/class_2248;Lnet/minecraft/class_2248;)V
+		ARG 1 bed
+		ARG 2 particleSource
+	METHOD method_25684 registerBell ()V
+	METHOD method_25685 registerTurnableRail (Lnet/minecraft/class_2248;)V
+		ARG 1 rail
+	METHOD method_25686 registerRoots (Lnet/minecraft/class_2248;Lnet/minecraft/class_2248;)V
+		ARG 1 root
+		ARG 2 pottedRoot
+	METHOD method_25687 registerGrindstone ()V
+	METHOD method_25688 registerStraightRail (Lnet/minecraft/class_2248;)V
+		ARG 1 rail
+	METHOD method_25689 registerCampfire ()V
 	METHOD method_25690 registerCommandBlock (Lnet/minecraft/class_2248;)V
-		ARG 1 block
+		ARG 1 commandBlock
+	METHOD method_25691 registerBookshelf ()V
 	METHOD method_25692 registerAnvil (Lnet/minecraft/class_2248;)V
+		ARG 1 anvil
+	METHOD method_25693 registerRedstone ()V
+	METHOD method_25694 registerMushroomBlock (Lnet/minecraft/class_2248;)V
+		ARG 1 mushroomBlock
+	METHOD method_25695 registerComparator ()V
+	METHOD method_25696 registerFurnaceLikeOrientable (Lnet/minecraft/class_2248;)V
+		ARG 1 block
+	METHOD method_25697 registorSmoothStone ()V
+	METHOD method_25698 registerNetherrackBottomCustomTop (Lnet/minecraft/class_2248;)V
+		ARG 1 block
+	METHOD method_25699 registerBrewingStand ()V
+	METHOD method_25700 getFireFloorModels (Lnet/minecraft/class_2248;)Ljava/util/List;
+		ARG 1 texture
+	METHOD method_25701 registerCake ()V
+	METHOD method_25702 getFireSideModels (Lnet/minecraft/class_2248;)Ljava/util/List;
+		ARG 1 texture
+	METHOD method_25703 registerCartographyTable ()V
+	METHOD method_25704 getFireUpModels (Lnet/minecraft/class_2248;)Ljava/util/List;
+		ARG 1 texture
+	METHOD method_25705 registerSmithingTable ()V
+	METHOD method_25706 registerLantern (Lnet/minecraft/class_2248;)V
+		ARG 1 lantern
+	METHOD method_25707 registerPumpkins ()V
+	METHOD method_25708 registerNorthDefaultHorizontalRotation (Lnet/minecraft/class_2248;)V
 		ARG 1 block
+	METHOD method_25709 registerCauldron ()V
+	METHOD method_25710 registerShulkerBox (Lnet/minecraft/class_2248;)V
+		ARG 1 shulkerBox
+	METHOD method_25711 registerChorusFlower ()V
+	METHOD method_25712 registerEndPortalFrame ()V
+	METHOD method_26279 registerRespawnAnchor ()V
+	CLASS class_4911 BuiltinBlockDerivative
+		FIELD field_22835 modelId Lnet/minecraft/class_2960;
+		METHOD <init> (Lnet/minecraft/class_4910;Lnet/minecraft/class_2960;Lnet/minecraft/class_2248;)V
+			ARG 2 modelId
+			ARG 3 block
+		METHOD method_25713 includeWithItem (Lnet/minecraft/class_4942;[Lnet/minecraft/class_2248;)Lnet/minecraft/class_4910$class_4911;
+			ARG 1 model
+			ARG 2 blocks
+		METHOD method_25714 includeWithItem ([Lnet/minecraft/class_2248;)Lnet/minecraft/class_4910$class_4911;
+			ARG 1 blocks
+		METHOD method_25715 includeWithoutItem ([Lnet/minecraft/class_2248;)Lnet/minecraft/class_4910$class_4911;
+			ARG 1 blocks
+	CLASS class_4912 DecorativeDerivative
```

Comment:
This seems to include things like buttons, signs, fences, pressure plates... Those aren't really decorative. Also, why is this not `DecorativeBlockDerivative`, like `BuiltinBlockDerivative`?

Reply:
This one generates block states and models from a base block's texture. Dunno what a better name could be. If this is not `Decorative`, what name do you suggest?

Reply:
Done. now called block texture pool